src/pool/cli.c

```diff
@@ -2101,6 +2141,30 @@ dc_pool_set_attr(tse_task_t *task)
 
 	in = crt_req_get(cb_args.pra_rpc);
 	in->pasi_count = args->n;
+
+	/* no easy way to determine if a name storage address is likely
+	 * to cause an EFAULT during memory registration, so duplicate
+	 * name in heap
+	 * XXX may need to do the same for values ??
+	 */
+	D_ALLOC_ARRAY(new_names, args->n);
+	if (!new_names)
+		D_GOTO(out, rc = -DER_NOMEM);
+	for (i = 0 ; i < args->n ; i++) {
+		D_STRNDUP(new_names[i], args->names[i], DAOS_ATTR_NAME_MAX);
```

Comment:
this truncates the user attribute name, right?

Reply:
no it should not, because with change in attr_check_input() attribute name will not exceed DAOS_ATTR_NAME_MAX

Reply:
ah ok, makes sense.